src/server/room-controller.js

```diff
@@ -89,6 +90,25 @@ export default class RoomController {
     return res.send({ rooms: mappedRooms });
   }
 
+  async handleGetRoom(req, res) {
+    const { user } = req;
+    const { roomId } = req.params;
+
+    const room = await this.roomService.getRoomById(roomId);
+
+    if (!room) {
+      throw new NotFound();
+    }
+
+    if (!isRoomOwnerOrInvitedCollaborator({ room, userId: user._id })) {
```

Comment:
I would change this to `!isRoomOwnerOrInvitedMember`. I did see that this is intentional after checking the tests, but neither the endpoint path nor the `handleGetRoom` handler suggest why this endpoint is dedicated to collaborators only. I think we don't have any reason to exclude members of `exclusive` rooms here actually.

We should also call `const mappedRoom = await this.clientDataMappingService.mapRoom(room, isRoomOwner ? req.user : null);` like we do when we get the page, so that the users still have access to the same members data after this API call is made.

Reply:
Good catch! Changed both. It took me quite a long time what the ternary operator is for in `isRoomOwner ? req.user : null`, so I moved this decision into the mapper where I think it's easier to grasp.

Reply:
Cool 👍